gui/device_list_frame.py

```diff
@@ -119,9 +113,9 @@ def set_on_del_click(self, on_click=lambda self: self):
     def set_on_display_click(self, on_click=lambda self: self):
 
         def callback():
-            device = adb.bridge.get_device(self.ip, self.port)
-            device.name = self.name
-            device.save_file_prefix = f"{self.name}_{device.serial.replace(':', '_')}"
+            device = adb.bridge.get_device(self.serial_no)
+            device.name = self.serial_no
+            device.save_file_prefix = f"{self.serial_no}"
```

Comment:
path file cannot include colon, so if serial_no is 127.0.0.1:5555 there is a bug here.

Reply:
@yuceltoluyag pls check

Reply:
can give me log

Reply:
@lilyuppi can give me log

Reply:
I just saw this, can you show the error output ? I stopped using Bluestacks years ago, are you using bluestacks ? Because in LDPlayer I can see the emulator directly without having to enter ip, port.

Reply:
I don't know, will you see this notification, @ndhao164 ?



A long time ago a friend of mine told me that the game doesn't create files in the save folder. I had a chance to look at it today, he was using Bluestacks. When I looked in the save folder, a file called 127.0.0.0.1 was being created and 127.0.0.1_building.json was not being created, which it should normally be. Actually, the file that should be created in the save folder is 127.0.0.1_config.json, but I realized that it was not created because of the editing here. I created this repo about 1 year ago, so someone must have used a different emulator. Actually, people who play games on emulators, especially those with low-end PCs, definitely don't use Bluestacks.



I solved his problem, @lilyuppi  is right here, it doesn't return an error message but it doesn't fulfill its function either. I fixed it but didn't have a chance to post it to the repo. I was using LDPlayer when this happened, I had this comment in mind, just to remind you that this is an incorrect usage. In LDPlayer this is not noticeable, but in some emulators the naming is not done as in LDPlayer. For example, in LDPlayer the device name is "emulator-5554", in other emulators it might be "127.0.0.1:25015". This prevents usage.